lib/util/source-code-util.js

```diff
@@ -25,14 +29,19 @@ var assign = require("object-assign"),
  * Get the SourceCode object for a single file
  * @param   {string}     filename The fully resolved filename to get SourceCode from.
  * @param   {Object}     options  A CLIEngine options object.
- * @returns {SourceCode}          The SourceCode object representing the file.
+ * @returns {Array}               Array of the SourceCode object representing the file
+ *                                and fatal error message.
  */
 function getSourceCodeOfFile(filename, options) {
+    debug("getting sourceCode of", filename);
     var opts = assign({}, options, { rules: {}});
     var cli = new CLIEngine(opts);
-    cli.executeOnFiles([filename]);
+    var results = cli.executeOnFiles([filename]);
+    if (results && results.results[0] && results.results[0].messages[0] && results.results[0].messages[0].fatal) {
+        return [null, results.results[0].messages[0].message];
+    }
     var sourceCode = eslint.getSourceCode();
-    return sourceCode;
+    return [sourceCode, null];
```

Comment:
You need to use a try-catch whjen you call this function and then pass the error to cb().